v2/controllers/blockrequest_controller.go

```diff
@@ -47,7 +47,23 @@ func (r *BlockRequestReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, nil
 	}
 
-	block, err := r.Manager.AllocateBlock(ctx, br.Spec.PoolName, br.Spec.NodeName)
+	blocks := &coilv2.AddressBlockList{}
+	err = r.Client.List(ctx, blocks, client.MatchingFields{
```

Comment:
You may be able to use LabelSelector instead of MatchingFields.
LabelSelector is simple. However, MatchingFields is faster because it indexes.

Reply:
I think `MatchingFields` is no problem.